contracts/utils/cryptography/EIP712.sol

```diff
@@ -131,12 +131,34 @@ abstract contract EIP712 is IERC5267 {
     {
         return (
             hex"0f", // 01111
-            _name.toStringWithFallback(_nameFallback),
-            _version.toStringWithFallback(_versionFallback),
+            _EIP712Name(),
+            _EIP712Version(),
             block.chainid,
             address(this),
             bytes32(0),
             new uint256[](0)
         );
     }
+
+    /**
+     * @dev The name parameter for the EIP712 domain.
+     *
+     * NOTE: This function reads from storage by default, but can be redefined to return a constant value if gas costs
```

Comment:
A few things:
- By default we read `_name` with is an immutable value. We only read from storage if necessary (in case the value is too large to fit in a ShortString.
- Since this is not virtual, its not possible to redefine it to return a constant value.

Reply:
I've updated the comments